### PR TITLE
Feature/cloudinary

### PR DESCRIPTION
**Pull request by Shantanu1605**
Fixed the Cloudinary feature

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,11 +12,14 @@
   "author": "Sabeel Khan",
   "license": "ISC",
   "dependencies": {
+    "cloudinary": "^1.39.0",
     "dotenv": "^16.3.1",
     "ejs": "^3.1.9",
     "ejs-mate": "^4.0.0",
     "express": "^4.18.2",
-    "mongoose": "^7.3.4"
+    "mongoose": "^7.4.1",
```
Comment (sabeelkhan99) [inline package.json:20]: revert this to `7.3.4`

File: src/app.js [hunk 2]
```diff
@@ -16,4 +17,5 @@ app.use(express.urlencoded({ limit: '10mb', extended: true, parameterLimit: 5000
 app.use(healthcheckRoutes);
 app.use('/api/v1', v1Routes);
 
+
```
Comment (sabeelkhan99) [inline src/app.js:20]: Extra line

File: src/controllers/productController.js [hunk 3]
```diff
@@ -1,24 +1,46 @@
+const dotenv = require('dotenv');
```
Comment (sabeelkhan99) [inline src/controllers/productController.js:1]: Not required env variable available via configs

File: src/controllers/productController.js [hunk 4]
```diff
@@ -1,24 +1,46 @@
...
-        price: req.body.price,
-        desc: req.body.desc,
-        imageUrl: req.body.imageUrl,
-        quantity: req.body.quantity,
-        rating: req.body.rating,
-    };
-
-    const newProduct = await productService.create(product);
-    res.status(201).json(newProduct);
+  try{
```
Comment (sabeelkhan99) [inline src/controllers/productController.js:14]: Remove try catch use `catchAsync` util method.

File: src/controllers/productController.js [hunk 5]
```diff
@@ -1,24 +1,46 @@
...
+    console.log(req.file);
+    const result = await cloudinary.uploader.upload(req.file.path);
+
+  console.log(result);
```
Comment (sabeelkhan99) [inline src/controllers/productController.js:18]: remove `console.log`

File: src/controllers/productController.js [hunk 6]
```diff
@@ -1,24 +1,46 @@
...
+  const newProduct = await productService.create(product);
+  res.redirect('/api/v1/products');
+}catch(err){
+  console.log(err);
+}
+
 };
 
+ 
+
```
Comment (sabeelkhan99) [inline src/controllers/productController.js:39]: Extra line

File: src/controllers/productController.js [hunk 7]
```diff
@@ -1,24 +1,46 @@
...
+const addProduct = (req,res)=>{
+    res.render('products/showNewForm');
+}
```
Comment (sabeelkhan99) [inline src/controllers/productController.js:42]: Already in `env/sbox`

File: src/controllers/productController.js [hunk 8]
```diff
@@ -29,4 +51,5 @@ module.exports = {
     getAllProducts,
     create,
     findById,
+    addProduct
```
Comment (sabeelkhan99) [inline src/controllers/productController.js:54]: same as : [Link](https://github.com/sabeelhps/CB-Jan23-E-Commerce/pull/38/files#r1292201704)

File: src/models/Product.js [hunk 9]
```diff
@@ -17,4 +18,5 @@ const productSchema = new mongoose.Schema({
 
 const Product = mongoose.model('Product', productSchema);
 
+
```
Comment (sabeelkhan99) [inline src/models/Product.js:21]: Extra line

File: src/routes/v1/productRoutes.js [hunk 10]
```diff
@@ -3,9 +3,14 @@ const express = require('express');
...
 const productController = require('../../controllers/productController');
 
+const upload = require("../../utils/multer");
+  
 router.get('/', productController.getAllProducts);
 
-router.post('/', productController.create);
+router.post('/',upload.single("imageUrl"), productController.create);
+
+
```
Comment (sabeelkhan99) [inline src/routes/v1/productRoutes.js:12]: Extra line

File: src/utils/multer.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+
+module.exports = multer({
+    // dest: "./upload",
+    storage: multer.diskStorage({}),
+    fileFilter: (req,file,cb)=>{
+        // let ext = path.extname(file.originalname);
+        // if(ext!== ".jpg" && ext !==".jpeg" ){
+        //     cb(new Error("Unsupported file type!"), false);
+        //     return;
+        // }
```
Comment (sabeelkhan99) [inline src/utils/multer.js:12]: Remove commments if not required

**Comment by kd-labs** [comment]
Closing PR as feature to upload images to cloudinary is merged via another PR

**Pull request closed without merging**